### ISSUE TITLE
Using a default profile with --keep-profile-changes will break Firefox

### ISSUE THREAD

**Issue by kumar303** [mozilla/web-ext#1005]
### Is this a feature request or a bug?

bug

### What is the current behavior?

Consider a command like this (DO NOT run this):

```
web-ext run --firefox-profile=default --keep-profile-changes
```

### What is the expected or desired behavior?

Here's what web-ext will do to your default profile:
- It will disable automatic updates to Firefox
- It will disable updates to all add-ons
- It will disable the prompt for accepting a remote debugging connection -- this means anyone can silently connect a remote debugger to your Firefox
- It will disable signature checks for add-ons (but only in pre-release builds of Firefox)
- It will allow side loading add-ons which is how a lot of crapware installers inject their add-ons without user consent

This really shouldn't be possible at all. It should raise an error. We could possibly allow it with an option like `--allow-web-ext-to-completely-break-my-profile` because it's possible that someone may not care about trashing their default profile? I don't know, I'm leaning to not allowing this at all.

### Version information (for bug reports)

* **Firefox version**: N/A
* **Your OS and version**: N/A
* Paste the output of these commands:

````
node --version && npm --version && web-ext --version
v6.10.3
3.10.10
master-7b390e45f3a998a57238ddb9f163da4a61dfb08a
````

**Comment by kumar303** [comment]
The fix that landed was to completely disallow any default profile with the `--keep-profile-changes` option. You will see an error.

**Comment by autonome** [comment]
@kumar303 what determines the "default" profile? i've created a new profile and am seeing this message.

eg:

1. create new profile in Nightly named "foo"
2. open it, everything looks fine
3. web-ext run -p foo --keep-profile-changes

**Comment by autonome** [comment]
This might be a side effect of #1309, or at least related.

**Comment by kumar303** [comment]
Hi Dietrich. I’m on full time baby duty (parental leave) until June but
@rpl can probably help out. Did you try with a full path to the foo profile
directory? There are some bugs with named aliases.

**Comment by rpl** [comment]
@autonome The available profiles names are discovered using the firefox-profile dependency (https://github.com/saadtazi/firefox-profile-js), which looks into the `profiles.ini` file from the user directory:

https://github.com/saadtazi/firefox-profile-js/blob/46aca6c72fc889dfdd7114e1aa796797a3b9cabb/lib/profile_finder.js#L11-L25

and

https://github.com/saadtazi/firefox-profile-js/blob/46aca6c72fc889dfdd7114e1aa796797a3b9cabb/lib/profile_finder.js#L52-L64

Then web-ext will consider a default profile (https://github.com/mozilla/web-ext/blob/532951fbcc54892ee0e9e38f002a012ce2ab40e9/src/firefox/index.js#L211):

- a profile named "default" or "dev-edition-default": https://github.com/mozilla/web-ext/blob/532951fbcc54892ee0e9e38f002a012ce2ab40e9/src/firefox/index.js#L194-L197

- or one that has Default=1 in its profiles.ini section: https://github.com/mozilla/web-ext/blob/532951fbcc54892ee0e9e38f002a012ce2ab40e9/src/firefox/index.js#L247-L270

My guess is that this last part is likely the one that is triggering the error.

@autonome could you attach here a "cleaned / anonymized" version of your profiles.ini?
(or even just the fragment related to the profile that is wrongly considered a default profile)

Thanks in advance.

**Comment by autonome** [comment]
Yeah after looking at profile.ini, it looks like the profile manager creates new profiles and sets them to default. Will send you mine over email.

**Comment by autonome** [comment]
To clarify: I see the same error if using a default or non-default profile.

So just using profile *name* as seen in profile manager UI (no path and no randomized directory name prefix + ".") is broken for any profile that I try, and shows that error.

**Comment by rpl** [comment]
Thanks @autonome 
now I definitely remember that we already looked into that issue before :-( 
(some additional details about the underlying reasons are described in a bit more details in https://github.com/mozilla/web-ext/issues/932#issuecomment-298638782).

It is tracked by #932 and we have two pull requests that were trying to fix it (and still prevent web-ext from messing by mistake with a default profile, mainly because of the changes to the preferences that web-ext needs and that are potentially dangerous for a user that is not completely aware of their side-effects, as @kumar303 described in this issue).

The first pull request that has been trying to fix that issue is #968, we extracted and merged part of it in #1007 to prevent the risks described by this issue, the rest of #968 is now included in #1149, which is not yet merged, and it still needs some additional tweaks, but it doesn't seem too far from being ready.
